modules/swagger-codegen/src/main/resources/csharp/Logger.mustache

```diff
@@ -0,0 +1,55 @@
+{{>partial_header}}
+using System;
+
+namespace {{packageName}}.Client {
+    public partial class Logger {
+        public Logger(string loggerName) {
+            Initialize(loggerName);
+        }
+
+        partial void Initialize(string loggerName);
+
+        public void Trace(string message) {
+            TracePartial(message);
+        }
+
+        partial void TracePartial(string message);
```

Comment:
I don't understand why public method are passed through to partial methods of the same name suffixed with the method being "partialed".  Why not just use partials:

```
partial void Trace(string message);
```

This has the added benefit of allowing the compiler to optimize away unimplemented trace method calls if the implementor hasn't provided an implementation. The way this class is currently implemented, I believe all calls are executed as noops if the implementor hasn't provided the *Partial implementations.

I don't think the Logger class needs to be public, if that's a factor into the design here.